lib/jobs/redfish-discovery.js

```diff
@@ -0,0 +1,189 @@
+// Copyright 2016, EMC, Inc.
+
+'use strict';
+
+var di = require('di');
+var redfish = require('redfish-node');
+
+module.exports = RedfishDiscoveryJobFactory;
+di.annotate(RedfishDiscoveryJobFactory, new di.Provide('Job.Redfish.Discovery'));
+di.annotate(RedfishDiscoveryJobFactory,
+    new di.Inject(
+        'Job.Base',
+        'Logger',
+        'Promise',
+        'Assert',
+        'Util',
+        'Services.Waterline',
+        'Services.Encryption',
+        '_'
+    )
+);
+function RedfishDiscoveryJobFactory(
+    BaseJob,
+    Logger,
+    Promise,
+    assert,
+    util,
+    waterline,
+    encryption,
+    _
+) {
+    var logger = Logger.initialize(RedfishDiscoveryJobFactory);
+
+    /**
+     * @param {Object} options task options object
+     * @param {Object} context graph context object
+     * @param {String} taskId running task identifier
+     * @constructor
+     */
+    function RedfishDiscoveryJob(options, context, taskId) {
+        RedfishDiscoveryJob.super_.call(this,
+                                   logger,
+                                   options,
+                                   context,
+                                   taskId);
+
+        assert.object(this.options);
+        assert.string(this.options.uri);
+        this.settings = {
+            uri: this.options.uri,
+            username: this.options.username,
+            password: this.options.password
+        };
+        
+        var apiClient = new redfish.ApiClient();
+        apiClient.basePath = this.settings.uri.replace(/\/+$/, '');
+        
+        // setup basic authorization
+        if (!_.isUndefined(this.settings.username) && 
+            !_.isUndefined(this.settings.password)) {
+            var token = new Buffer(
+                this.settings.username + ':' + this.settings.password
+            ).toString('base64');
+            apiClient.defaultHeaders.Authorization = 'Basic ' + token;
+            process.env.NODE_TLS_REJECT_UNAUTHORIZED = "0"; // (jl) need a ssl_verify option here.
+        }
+        this.redfishApi = Promise.promisifyAll(new redfish.RedfishvApi(apiClient));
+    }
+    util.inherits(RedfishDiscoveryJob, BaseJob);
+
+    /**
+     * @memberOf RedfishDiscoveryJob
+     */
+    RedfishDiscoveryJob.prototype._run = function() {
+        var self = this;
+        return Promise.resolve()
+        .then(function() {
+            return self.createChassis();
+        })
+        .then(function() {
+            return self.createSystems();
+        })
+        .catch(function(err) {
+            throw err;
```

Comment:
In the catch block you should do `self._done(err)` and in the final then block you should do `self._done()`.